code/eyetracking/edf-to-bids.ipynb

```diff
@@ -2,1437 +2,578 @@
-       "1               336.200012  26.5  26.5  ...     0.0  4.566832e-41     0.0   \n",
-       "2               335.700012  26.5  26.5  ...     0.0  4.566832e-41     0.0   \n",
-       "3               335.799988  26.5  26.5  ...     0.0  4.566832e-41     0.0   \n",
-       "4               336.399994  26.5  26.5  ...     0.0  4.566832e-41     0.0   \n",
-       "...                    ...   ...   ...  ...     ...           ...     ...   \n",
-       "2007887           0.000000   0.0   0.0  ...     0.0  0.000000e+00     0.0   \n",
-       "2007888           0.000000   0.0   0.0  ...     0.0  0.000000e+00     0.0   \n",
-       "2007889           0.000000   0.0   0.0  ...     0.0  0.000000e+00     0.0   \n",
-       "2007890           0.000000   0.0   0.0  ...     0.0  0.000000e+00     0.0   \n",
-       "2007891           0.000000   0.0   0.0  ...     0.0  0.000000e+00     0.0   \n",
-       "\n",
-       "         frxyvel        fryvel    flags    input  buttons    htype  errors  \n",
-       "0            0.0  4.566832e-41  32641.0  32768.0      0.0 -32768.0     0.0  \n",
-       "1            0.0  4.566832e-41  24449.0  32768.0      0.0 -32768.0     0.0  \n",
-       "2            0.0  4.566832e-41  24449.0  32768.0      0.0 -32768.0     0.0  \n",
-       "3            0.0  4.566832e-41  24449.0  32768.0      0.0 -32768.0     0.0  \n",
-       "4            0.0  4.566832e-41  24449.0  32768.0      0.0 -32768.0     0.0  \n",
-       "...          ...           ...      ...      ...      ...      ...     ...  \n",
-       "2007887      0.0  0.000000e+00      0.0      0.0      0.0      0.0     0.0  \n",
-       "2007888      0.0  0.000000e+00      0.0      0.0      0.0      0.0     0.0  \n",
-       "2007889      0.0  0.000000e+00      0.0      0.0      0.0      0.0     0.0  \n",
-       "2007890      0.0  0.000000e+00      0.0      0.0      0.0      0.0     0.0  \n",
-       "2007891      0.0  0.000000e+00      0.0      0.0      0.0      0.0     0.0  \n",
-       "\n",
-       "[2007892 rows x 27 columns]"
-      ]
-     },
-     "execution_count": 92,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
    "source": [
-    "samples = samples.reindex(columns=[c for c in samples.columns if \"left\" not in c])\n",
-    "samples.rename(columns={\"time\": \"eye_timestamp\", \"gx_right\": \"eye1_x_coordinate\", \"gy_right\": \"eye1_y_coordinate\", \"pa_right\": \"eye1_pupil_size\", \"px_right\": \"eye1_pupil_x_coordinate\", \"py_right\": \"eye1_pupil_y_coordinate\"})"
+    "Now, as an illustration, let's plot the participant's pupil size during the dwi run. The gaps in the plot directly correlate to moments when the participant closed their eyes."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 93,
-   "id": "18e58291",
+   "execution_count": null,
+   "id": "4c2bfcf4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DwiSession4.plot_pupil_size()"
```

Comment:
I had an issue running this cell (and the next) saying:
``` python
ValueError: Multi-dimensional indexing (e.g. `obj[:, None]`) is no longer supported. Convert to a numpy array before indexing instead.
```
I believe this comes from my version of pandas/python (3.9) being more recent - I fixed it specifying `.values` to send numpy arrays to pyplot instead of pandas series (see comment on `eyetrackingrun.py`)

Reply:
Yes, indeed. Thanks for pointing that out. I used Python 3.6, as suggested in the pyedfread readme. But if it's stable with more recent versions, let's switch to the more recent python.